### DISCUSSION TITLE
「トランザクション分離」vs 「トランザクション隔離」

### DISCUSSION THREAD

**Discussion by koizumistr** [pgsql-jp/jpug-doc#2608] [NGとする語]
## 問題点

transaction isolationの訳は「トランザクション分離」か「トランザクション隔離」

## 背景

こちらも #1266 同様#1265 で話題になった件です。

## 解決方法

議論しましょう。

## 注意点

やはり統一はしたいですね。

**Comment by ester41** [comment]
各RDBMSの翻訳を見てみたところ、どこも「トランザクション分離」となっていました。
「分離」に統一する方に一票です。

Oracle: http://otndnld.oracle.co.jp/document/products/oracle11g/111/doc_dvd/server.111/E05765-03/consist.htm
SQLServer: https://www.microsoft.com/ja-jp/language/Search?&searchTerm=transaction%20isolation&langID=%E6%97%A5%E6%9C%AC%E8%AA%9E&Source=true&productid=%E3%81%99%E3%81%B9%E3%81%A6%E3%81%AE%E8%A3%BD%E5%93%81
MySQL: https://dev.mysql.com/doc/refman/5.6/ja/dynindex-isolevel.html

**Comment by koizumistr** [comment]
以下のような状況でこの文書内では「トランザクション隔離」が優勢ですね。

$grep "トランザクション分離" `doc/src/sgml/*.sgml` `doc/src/sgml/ref/*.sgml` | wc
      17      45    2920
$grep "トランザクション隔離" `doc/src/sgml/*.sgml` `doc/src/sgml/ref/*.sgml` | wc
      29      66    7226

が、「分離」でも「隔離」でもない訳をしているものもあるような、、、、

$grep "transaction isolation" `doc/src/sgml/*.sgml doc/src/sgml/ref/*.sgml` | wc
      53     445    5160

**Comment by koizumistr** [comment]
「トランザクション隔離」で検索すると、上位2件が自分たち、その下はほとんどが「トランザクション分離」で出てきますので、私も「トランザクション分離」に一票です。

![default](https://user-images.githubusercontent.com/10354396/41190968-90cbe37e-6bd7-11e8-8cc5-035377fe8331.png)

**Comment by koizumistr** [comment]
ちなみに「の」を入れると以下のような状況です。

$grep "トランザクションの隔離" `doc/src/sgml/*.sgml doc/src/sgml/ref/*.sgml` | wc
       3      10     708
$grep "トランザクションの分離" `doc/src/sgml/*.sgml doc/src/sgml/ref/*.sgml` | wc
       7      17    1486

**Comment by tatsuo-ishii** [comment]
> 「分離」でも「隔離」でもない訳
たまたま見つけましたが、ref/set_transaction.sgmlにそういうのがありますね。"isolation level"がすっぽり訳抜けしています。

**Comment by KenichiroTanaka** [comment]
同じくトランザクション分離、に一票です。wikiやデータベーススペシャリスト試験本対策本でもそのようになっています。

隔離とすべき、という意見はありますでしょうか。
無ければ分離で行きたいと思います。

**Comment by tatsuo-ishii** [comment]
JIS規格では「トランザクション隔離性水準」です。

**Comment by noborus** [comment]
隔離 → 分離であれば、一括変換でやっても問題なさそうです。
pull requestを送っておきます。
入れてからリリースしても良いですし、見送っても良いです。

**Comment by KenichiroTanaka** [comment]
PRもいただいているのでそろそろ決めたいと思います。

A.広く一般的に使われている「トランザクション分離レベル」
B.JISに合わせた「トランザクション隔離レベル」
かと思います。

広く使われている方が読者も理解しやすい、検索数からも方言や誤った翻訳であると言えないかと思いますので
A.が良いように思います。皆様いかがでしょうか。

**Comment by noborus** [comment]
NGリストにトランザクション隔離レベルを追加するPRを出しました。 #2607
